### ISSUE TITLE
Audio Issues while on VPN - and (hacky?) fix

### ISSUE THREAD

**Issue by Andy-Roger-Kats** [mozilla/hubs#4584]
**Description**
We deployed the latest hubs-cloud branch to our server, and users who join the room through our company VPN are having audio issues. In many cases, no users can hear each other, and in other cases one user can hear the other but not vice versa. This issue is not consistently reproduceable and happens about 20 - 30% of the time on average. 

**To Reproduce**
Steps to reproduce the behavior: - this repro is using our companies specific VPN setup so just explaining here how we accomplished the reproduction of the issue several times. 
1. Users will test on a domain that has latest (current) Hubs Cloud branch deployed.
2. In this series of tests, user A was located in NYC and user B in WA, user A was connected though a VPN server located in NJ, the user B was connected through a VPN located in CA - both VPNs are provided by the same company.
3. Open a Chrome incognito browser (using version 92.0 incognito mode).
4. Both users enter room via a room share code.
5. Both users make sure audio is set up correctly and works.
6. Both users enter the room.
7. The users move their avatars close to one another and take turns saying "check 1, 2" (the users are also connected via phone so they can coordinate if the issue occurs).
8. If there is an issue with audio on either side have both users try:  `NAF.connection.adapter._recvTransport.connectionState` and `NAF.connection.adapter._sendTransport.connectionState` in their consoles to see which command / user logs "disconnected"
9. Try the hack fix, on the user that has a "disconnected" log, have them add this to their console: ` NAF.connection.adapter.reconnect()` to see if the audio starts to work again.

**Expected behavior**
_A clear and concise description of what you expected to happen._
On a VPN setup similar to the company VPN we are using in these tests we expect users not be able to hear each other 20 - 30% of the time. If the command in step 9. is run on the user with a "disconnected" log (see step 8.) the users should be able to start hearing each other.

**Hardware** 
 - Device: PC (Dell - did not get model), 2019 MacBook Pro2.4GHz Intel Code i5
 - OS: Windows / Mac
 - Browser: All tests ran on Chrome 92.0 incognito mode

**Additional context**
The inconsistency of reproducing the issue leads me to think it was caused by a race-condition during the WebRTC connection and something about the VPN setup at our company causing the WebRTC channels to not connect. When there is an issue the users are able to check the console with `NAF.connection.adapter._recvTransport.connectionState` and `NAF.connection.adapter._sendTransport.connectionState` and one or the other (on one or both of the users) will log "disconnected". 

For example if user A cannot hear anyone either `NAF.connection.adapter._recvTransport.connectionState` or user B's `NAF.connection.adapter._sendTransport.connectionState` will log "disconnected". These logs seem to be directly related to the issue.  Interestingly, when there are audio issues, if the user with either or both `_recvTransform` /`_sendTransport` is in the "disconnected" state the user can run the `NAF.connection.adapter.reconnect()` and it fixes the issue.

Question:
1. Do you have any guidance on what ports would need to be unblocked on a company firewall? I have heard that UDP 5060 is used for unencrypted and UDP 5061 for encrypted. Sorry if this is a misguided question but I don't normally deal with network admin - but any help with how I can solve for this on our end would be very helpful!
2. Is the hack fix in step 8. of the repro a viable fix? Should be write some code that checks for the "disconnected" state then run the `reconnect()` function? Are their side effects to this that we should take into account?

I initially commented on this here: https://github.com/mozilla/hubs/issues/3096 but I thought I would open a new issue with new test data,  addition context, and questions I had. Please consolidate with that issue if you think it is best!

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/HUBS-768)
